deploy/examples/cert-manager/smcp-2.3.1.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: maistra.io/v2
```

Comment:
> Why do we need a separate SMCP manifest for 2.3.1?

Because 2.3.0/1 need fields `pilotSecretName` and `rootCAConfigMapName`. In 2.3.2 the first will be ignored, and the second will be optional.

> would renaming it to smcp-2.3.x.yaml be more appropriate?

I don't think so, because in 2.3.2+ `pilotSecretName` and `rootCAConfigMapName` will be not recommended to use.